tests/test_denest.py

```diff
@@ -184,3 +184,87 @@ def test__schema__nested_arrays_add_tables():
 
         print('PASSED')
         print()
+
+
+def test__records__nested():
+    denested = denest.to_table_batches({
+        "properties": {
+            "a": {"type": "object",
+                  "properties": {
+                      "b": {
+                          "type": "array",
+                          "items": {
+                              "type": "object",
+                              "properties": {
+                                  "c": {
+                                      "type": "object",
+                                      "properties": {
+                                          "d": {"type": "integer"},
+                                          "e": {"type": "array",
+                                                "items": {"type": "object",
+                                                          "properties": {
+                                                              "f": {"type": "string"},
+                                                              "g": {"type": "boolean"}}}}}}}}}}}}},
+        [],
+        [{"a": {"b": []}},
+         {"a": {"b": [{"c": {"d": 1}}]}},
+         {"a": {"b": [{"c": {"d": 12}},
+                      {"c": {"d": 123}}]}},
+         {"a": {"b": [{"c": {"d": 1234}},
+                      {"c": {"d": 12345}},
+                      {"c": {"d": 123456}}]}},
+         {"a": {"b": [{"c": {"e": [{"f": "hello",
+                                    "g": True},
+                                   {"f": "goodbye",
+                                    "g": True}]}}]}}])
+
+    print('denested:', denested)
+
+    assert 3 == len(denested)
+
+    root_table_checked = False
+    nested_table_checked = False
+    nested_nested_table_checked = False
```

Comment:
Should we split up the test in three parts like this in the same test? We could place these three in separate test functions or just have one loop through `denested` in one test function in my opinion. What do you think is the way to go here?

Reply:
I like that idea. I'll refactor for that. My main concern originally with going that route, was that there'll be a modicum of "helper" code around these tests which will be obfuscating the actual test. Additionally, I was worried that previously we did _really_ precise tests which looked for specific things and hence had specific schemas. The bug we're experiencing is because we didn't have a gnarlier schema and records to pick apart etc. As such, I'm going to plan on the three tests sharing the same schema and records.

That all sound right to you?

Reply:
Sounds perfect!